### PR TITLE
[FLINK-27640][Connector/Hive] Exclude Pentaho dependency from Hive

### PR DESCRIPTION
**Pull request by MartijnVisser**
## What is the purpose of the change

* Exclude `org.pentaho:pentaho-aggdesigner-algorithm` from the Hive dependencies to avoid users with newer Maven versions having issues to build Flink locally. 

## Brief change log

* Excluded `org.pentaho:pentaho-aggdesigner-algorithm` in Hive connector pom.xml

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-connectors/flink-connector-hive/pom.xml
Comment (MartijnVisser) [inline flink-connectors/flink-connector-hive/pom.xml:868]: While it was excluded here, there is no Pentaho included in `hive-webhcat-java-client` according to the dependency tree

**Comment by MartijnVisser** [comment]
@luoyuxia Let me know what you think

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Comment by luoyuxia** [comment]
Theses change looks good to me. I perfer to find some times to verify it will still work in my hive test env.

**Review by gunnarmorling — COMMENTED** [review]
Hey @MartijnVisser, thanks for opening this PR. Stumbled upon this very same issue [just recently](https://www.morling.dev/blog/maven-what-are-you-waiting-for/). I can confirm that the dependency is not part of the tree any longer as per your change; are we sure though it's really not needed by Hive? If we are, then this LGTM.

**Comment by MartijnVisser** [comment]
> Hey @MartijnVisser, thanks for opening this PR. Stumbled upon this very same issue [just recently](https://www.morling.dev/blog/maven-what-are-you-waiting-for/). I can confirm that the dependency is not part of the tree any longer as per your change; are we sure though it's really not needed by Hive? If we are, then this LGTM.

Prior to opening the PR I've asked one of our Hive connector maintainers @luoyuxia if it's needed by Hive. From what I could find Pentaho is rather old report creation tool; Flink definitely doesn't do anything with it

**Comment by MartijnVisser** [comment]
> Theses change looks good to me. I perfer to find some times to verify it will still work in my hive test env.

Let's leave it open until you have been able to verify it

**Comment by rmetzger** [comment]
@MartijnVisser I hope its fine that I asked @gunnarmorling to take a look at this change :) 
+1 to wait until we know that Hive still works.

**Comment by luoyuxia** [comment]
> Theses change looks good to me. I perfer to find some times to verify it will still work in my hive test env.

I'm back. I have verified with Hive 2.3.9 & Hive 3.1.3, it still works after we exclude pentaho dependency.

**Comment by MartijnVisser** [comment]
> I'm back. I have verified with Hive 2.3.9 & Hive 3.1.3, it still works after we exclude pentaho dependency.

Thanks for checking! 

> `mvn dependency:tree` tells that there is another dependency on `org.pentaho:pentaho-aggdesigner-algorithm` in `flink-sql-client`
> shouldn't it also be excluded there?

Yes we should. I'll update the PR to include that, thanks for double checking.

**Comment by MartijnVisser** [comment]
> @MartijnVisser I hope its fine that I asked @gunnarmorling to take a look at this change :)
> +1 to wait until we know that Hive still works.

@rmetzger If you can have another look, I can finish this PR

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Comment by rmetzger** [comment]
Looks like the CI failure is a known issue: https://issues.apache.org/jira/browse/FLINK-29427

**Review by rmetzger — APPROVED** [review]
The change lgtm +1 to merge

**Pull request merged**
